src/main/scala/org/apache/spark/deploy/armada/submit/ConfigGenerator.scala

```diff
@@ -38,10 +38,11 @@ private[submit] class ConfigGenerator(val prefix: String, val conf: SparkConf) {
 
   private val confFiles = getConfFiles
   private def getConfFiles: Array[File] = {
+    val VALID_FILE_NAME_REGEX = "([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]"
```

Comment:
I think you may be able to replace the `A-Za-z0-9` portions of this with the `\w` character class shortcut - see https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html